src/Components/Canvas/Canvas.tsx

```diff
@@ -67,19 +87,21 @@ function sketch (p) {
             sketchData.points = [];
             reset = false;
             setClearCanvasInParent();
+            localStorage.removeItem("savedFigs");
         }
 
         p.mouseClicked = function (event) {
-            // if (event.type == 'touchstart') {
-            return Animation.mousePressed(sketchData, p);
+            Animation.mousePressed(sketchData, p);
+            return false;
         }
         
         p.mouseReleased = function() {
             Animation.mouseReleased(sketchData, p);
             // return false;
         }
-        
+
         Animation.draw(sketchData, p);
+        localStorage.setItem("savedFigs", JSON.stringify(sketchData.figs));
```

Comment:
It doesn't lag too much for now, but this really shouldn't have to be here. I'll approve but lets continue to search for ways to do this when the sketch dismounts. If that doesn't seem possible, then lets try to do this asynchronously.